src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -400,16 +397,12 @@ public void setBrowser(GitRepositoryBrowser browser) {
         }
         if (webUrls.size() == 1) {
             String url = webUrls.iterator().next();
-            if (url.startsWith("https://bitbucket.org/")) {
-                return new BitbucketWeb(url);
+            ExtensionList<BrowserGuesser> browserExtensions = BrowserGuesser.all();
+            Iterator<BrowserGuesser> itr = browserExtensions.iterator();
+            while(itr.hasNext()){
+                BrowserGuesser ext = itr.next();
+                return ext.guessBrowser(url);
             }
```

Comment:
If `ext.guessBrowser(url)` returns `null`, then this should continue the loop and try the remaining BrowserGuesser instances, rather than return `null` right away.

Reply:
It might make sense to move the loop into a static method in BrowserGuesser, similar to [SCMHead.HeadByItem.findHead](https://github.com/jenkinsci/scm-api-plugin/blob/400faa8fd5651731257ff59f53b715dfeede7ede/src/main/java/jenkins/scm/api/SCMHead.java#L269-L283). But I don't know what then would be good names for the abstract method and the static method.

Reply:
Oh yes. I forgot about other extensions in the extension list. Thank You :grin: .